client-js/TUTORIAL.md

```diff
@@ -0,0 +1,215 @@
+
+* `nftKind` - The `NFTKind` modality that specifies the off-chain items represented by the on-chain NFT data. This argument is passed in as a `u8` value and is required at the time of installation.
+
+* `jsonSchema` - The JSON schema for the NFT tokens that will be minted by the NFT contract passed in as a `String`. More information on `NFTMetadataKind` can be found [here](https://github.com/casper-ecosystem/cep-78-enhanced-nft#nftmetadatakind). This parameter may be left empty if metadata kind is set to `Raw(3)`. If the metadata kind is set to `CustomValidated(4)`, it will require a specifically formatted custom schema. This parameter **cannot be changed post installation**.
+
+* `nftMetadataKind` - The metadata schema for the NFTs to be minted by the NFT contract. This argument is passed in as a `u8` value and is required at the time of installation.
+
+* `identifierMode` - The `NFTIdentifierMode` modality dictates the primary identifier for NFTs minted by the contract. This argument is passed in as a `u8` value and is required at the time of installation.
+
+* `metadataMutability` - The `MetadataMutability` modality dictates whether the metadata of minted NFTs can be updated. This argument is passed in as a `u8` value and is required at the time of installation.
+
+* `mintingmode` - The `MintingMode` modality dictates the access to the `mint()` entry point in the NFT contract. This optional parameter will default to restricting access to the installer of the contract. **This parameter cannot be changed once the contract has been installed**.
+
+* `holdermode` - The `NFTHolderMode` modality dictates which entities can hold NFTs. This optional parameter will default to a mixed mode, allowing either `Accounts` or `Contracts` to hold NFTs. **This parameter cannot be changed once the contract has been installed**.
+
+* `burnMode` - The `BurnMode` modality dictates whether minted NFTs can be burned. This optional parameter will allow tokens to be burnt by default. **This parameter cannot be changed once the contract has been installed**.
+
+* `ownerReverseLookupMode` - The `OwnerReverseLookupMode` dictates whether the contract will index ownership of tokens as outlined [here](https://github.com/casper-ecosystem/cep-78-enhanced-nft#the-cep-78-page-system) to allow lookup of owned tokens by account. **This parameter cannot be changed once the contract has been installed**.
+
+Further information on CEP-78 modality options can be found in the base [cep-78-enhanced-nft](https://github.com/ACStoneCL/cep-78-enhanced-nft) repository on GitHub.
+
+## Minting a Token
+
+The CEP-78 JS Client includes code to construct a deploy that will `Mint` a token, as follows:
+
+```js
+
+  const mintDeploy = cc.mint(
+    {
+      owner: FAUCET_KEYS.publicKey,
+      meta: {
+        color: "Blue",
+        size: "Medium",
+        material: "Aluminum",
+        condition: "Used",
+      },
+    },
+    { useSessionCode },
```

Comment:
so this can be `{ useSessionCode: true }` or `{ useSessionCode: false }` depending on the `ownerReverseLookupMode` - `false` for `OwnerReverseLookupMode.NoLookup` and `true` for `OwnerReverseLookupMode.Complete`. If you're not sure how your contract is configured you can easily check this using this function `const ownerReverseLookupModeSetting = await cc.getReportingModeConfig();`